packages/frontend/src/app/events/page.tsx

```diff
@@ -0,0 +1,75 @@
+'use client';
+
+import { useSearchParams } from 'next/navigation';
+import React, { useEffect, useState } from 'react';
+
+import PaginationControls from '@/components/atoms/PaginationControls';
+import EventsDisplay from '@/components/molecules/EventsDisplay';
+import EventSearchPanel from '@/components/molecules/EventSearchPanel';
+import { EventService } from '@/services/EventService';
+import { Event } from '@/types';
+
+const EventPage = () => {
+  const searchParams = useSearchParams();
+  const [events, setEvents] = useState<Event[]>([]);
+  const [totalItems, setTotalItems] = useState<number>(0);
+  const [error, setError] = useState<string | null>(null);
+
+  const [title, setTitle] = useState<string>('');
+  const [tags, setTags] = useState<string[]>([]);
+  const [type, setType] = useState<string>('');
+  const [page, setPage] = useState<number>(searchParams.get('page') ? Number(searchParams.get('page')) : 1);
+
+  const per_page = searchParams.get('per_page') ? Number(searchParams.get('per_page')) : 12;
+
+  useEffect(() => {
+    try {
+      new EventService().getEvents(type, tags, title, page, per_page).then(({ events }) => {
+        setEvents(events);
+        setTotalItems(events.length);
+      });
+    } catch (error) {
+      setError('Error fetching events');
+    }
+  }, [page, per_page, tags, title, type]);
+
+  const handleSearch = (filters: { title: string; tags: string[]; type: string }) => {
+    setTitle(filters.title);
+    setTags(filters.tags);
+    setType(filters.type);
+  };
+
+  const handlePageChange = (newPage: number) => {
+    const url = new URL(window.location.href);
```

Comment:
accessing DOM properties directly is not best practice when working with react, we can use the useRouter hook that will achieve the same and be best practice https://nextjs.org/docs/pages/api-reference/functions/use-router